silkworm/db/datastore/snapshots/inverted_index_cache.hpp

```diff
@@ -0,0 +1,37 @@
+/*
+   Copyright 2025 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <utility>
+
+#include <silkworm/db/datastore/common/timestamp.hpp>
+
+#include "../common/entity_name.hpp"
+#include "common/cache.hpp"
+
+namespace silkworm::snapshots {
+
+struct InvertedIndexCacheData {
+    datastore::Timestamp requested;
+    datastore::Timestamp found;
+};
+using InvertedIndexCache = Cache<InvertedIndexCacheData>;
+using InvertedIndexCaches = std::map<datastore::EntityName, std::unique_ptr<InvertedIndexCache>>;
```

Comment:
I'd be more specific here: InvertedIndexSeekCache, because InvertedIndex in general might have more cacheable queries